### PR TITLE
added Tests for if9/authenticate.ts

### PR DESCRIPTION
**Pull request by Justus1994**

### CODE REVIEW

File: src/ifop/clients.ts [hunk 1]
```diff
@@ -2,10 +2,20 @@ import {NextFunction, Request, Response} from "express";
...
 };
 
+export function initDeviceDB() {
+  for (const userName in userDB) {
+    if (userDB.hasOwnProperty(userName)) {
+      deviceDB[getKey(userName, userDB[userName].deviceID)] = true;
+    }
+  }
+}
+
```
Comment (wzr1337) [inline src/ifop/clients.ts]: why is this needed? i would like to hold user info only in ifas

Comment (Justus1994) [reply src/ifop/clients.ts]: That was just for the test setup, I'm gonna remove it.

File: src/if9/authenticate.spec.ts [hunk 2]
```diff
@@ -115,6 +113,9 @@ describe("TEST validateServiceRequest", () => {
     const req = new MockExpressRequest(rdlOptions);
     const res = new MockExpressResponse();
 
+    const spyVIN = jest.spyOn(clients, "isDeviceRegistered");
+    spyVIN.mockReturnValueOnce(true);
+
     authenticate.validateServiceRequest(req, res, next);
     expect(next).toHaveBeenCalled();
   });
```
Comment (Justus1994) [inline src/if9/authenticate.spec.ts]: I removed the `initDeviceDB` method from [src/ifop/clients.ts](https://github.com/wzr1337/if9simu/pull/11/commits/a3483b0854eef59b53673d5c3ddf1bc6b2d9fe46#diff-3bbb73a5164b0493187165f4dfdc30af) and instead mocked `isDeviceRegistered` to test the function properly.

**Comment by wzr1337** [comment]
Thanks a lot
